packages/core/src/hooks/utilities/useAutoScroller.ts

```diff
@@ -43,14 +47,18 @@ export function useAutoScroller({
 
     scrollContainer.scrollBy(scrollLeft, scrollTop);
   }, []);
+  const sortedScrollableAncestors = useMemo(
+    () => sortScrollableAncestors(scrollableAncestors),
+    [sortScrollableAncestors, scrollableAncestors]
+  );
 
   useEffect(() => {
-    if (disabled || !scrollableAncestors.length || !draggingRect) {
+    if (disabled || !scrollableAncestors.length || !pointerCoordinates) {
```

Comment:
I don't know if it's intended, but it looks possible that `sortScrollableAncestors` could remove elements. Does it make sense the check the sorted ancestors list instead?

```suggestion
    if (disabled || !sortedScrollableAncestors.length || !pointerCoordinates) {
```

Reply:
What if you accept a sort function instead?

Reply:
Semi related to this change... is (or will) there be a way to stop an element scrolling?
When I first saw this PR I was thinking I could use it to stop the window scrolling horizontally, which is an issue I'm facing now with a vertical-only scrolling list.

Reply:
This looks good to me and I agree on the sorting 👍